Optimize acceptHeaderValueForImageResource()

cdumez opened the pull request:
#### 92dc76268c23b59fe1a5ae9a46cdf46cf5bdb7ac
<pre>
Optimize acceptHeaderValueForImageResource()
<a href="https://bugs.webkit.org/show_bug.cgi?id=262051">https://bugs.webkit.org/show_bug.cgi?id=262051</a>

Reviewed by Said Abou-Hallawa.

* Source/WebCore/loader/cache/CachedResourceRequest.cpp:
(WebCore::appendAcceptHeaderValueForAdditionalSupportedImageMIMETypes):
(WebCore::appendAcceptHeaderValueForVideoImageResource):
(WebCore::acceptHeaderValueForImageResource):
(WebCore::acceptHeaderValueForWebPImageResource): Deleted.
(WebCore::acceptHeaderValueForAVIFImageResource): Deleted.
(WebCore::acceptHeaderValueForJPEGXLImageResource): Deleted.
(WebCore::acceptHeaderValueForHEICImageResource): Deleted.
(WebCore::acceptHeaderValueForAdditionalSupportedImageMIMETypes): Deleted.
(WebCore::acceptHeaderValueForVideoImageResource): Deleted.

Canonical link: <a href="https://commits.webkit.org/268432@main">https://commits.webkit.org/268432@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/79e1281cbf40a7cb36d9eee2c3f454d1b81b1004

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/19711 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/20133 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/20750 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/21605 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/18426 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/19946 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/23393 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20276 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/19997 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/19930 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/19933 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/17145 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/22459 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17113 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/17936 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/24220 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18180 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/18110 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22203 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/18708 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/15853 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/17861 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4703 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/22215 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/31/builds/18533 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

shallawa commented on Source/WebCore/loader/cache/CachedResourceRequest.cpp:
```diff
@@ -128,66 +128,44 @@ void CachedResourceRequest::setDomainForCachePartition(const String& domain)
 }
 
-static constexpr ASCIILiteral acceptHeaderValueForWebPImageResource()
+static inline void appendAcceptHeaderValueForAdditionalSupportedImageMIMETypes(StringBuilder& builder)
```
This is a really long name but I think it is okay. Maybe something like this can be used:

```
static inline void appendAdditionalSupportedImageMIMETypes(StringBuilder& acceptHeader)
```

cdumez committed:
Optimize acceptHeaderValueForImageResource()
https://bugs.webkit.org/show_bug.cgi?id=262051

Reviewed by Said Abou-Hallawa.

* Source/WebCore/loader/cache/CachedResourceRequest.cpp:
(WebCore::appendAcceptHeaderValueForAdditionalSupportedImageMIMETypes):
(WebCore::appendAcceptHeaderValueForVideoImageResource):
(WebCore::acceptHeaderValueForImageResource):
(WebCore::acceptHeaderValueForWebPImageResource): Deleted.
(WebCore::acceptHeaderValueForAVIFImageResource): Deleted.
(WebCore::acceptHeaderValueForJPEGXLImageResource): Deleted.
(WebCore::acceptHeaderValueForHEICImageResource): Deleted.
(WebCore::acceptHeaderValueForAdditionalSupportedImageMIMETypes): Deleted.
(WebCore::acceptHeaderValueForVideoImageResource): Deleted.

Canonical link: https://commits.webkit.org/268432@main
